tests/storage-vm

```diff
@@ -281,6 +281,12 @@ for poolDriver in $poolDriverList; do
         echo "==> Testing VM optimized export/import (while running to check config.mount is excluded)"
         ! lxc export v1 --quiet --compression=none - --optimized-storage | tar -tf - | grep -F config.mount || false
 
+        if [[ $(hasNeededAPIExtension vm_zfs_storage_disk_live_resize) && $(hasNeededAPIExtension vm_lvm_storage_disk_live_resize) && ( "${poolDriver}" == "zfs" || "${poolDriver}" == "lvm" ) ]]; then
```

Comment:
I'd simplify the gating logic to handle only 1 extension + driver combo. Also, I think that `lvm-thin` would also work?

```suggestion
        LIVE_GROWING_SUPPORTED=0
        if hasNeededAPIExtension vm_zfs_storage_disk_live_resize && [ "${poolDriver}" = "zfs" ]; then
          LIVE_GROWING_SUPPORTED=1
        elif hasNeededAPIExtension vm_lvm_storage_disk_live_resize; then
          if [ "${poolDriver}" = "lvm" ] || [ "${poolDriver}" = "lvm-thin" ]; then
            LIVE_GROWING_SUPPORTED=1
          fi
        fi
```
The `$()` around `hasNeededAPIExtension vm_zfs_storage_disk_live_resize` is not needed as we only care about the function's return code, not its output.

Reply:
Great suggestion - thanks! And yes, `lvm-thin` should also work.